plugins/src/hashlookup/hashlookup.go

```diff
@@ -0,0 +1,492 @@
+				if errm != nil {
+					return nil, fmt.Errorf("Could not serialize: %s", err.Error())
+				}
+				tmpReq, err = http.NewRequest("POST", tmpUrl, bytes.NewReader(tmpBody))
+			} else {
+				tmpUrl = p.url + "/lookup/" + possibleSearchField + "/" + data.Get(possibleSearchField)
+				tmpReq, err = http.NewRequest("GET", tmpUrl, nil)
+			}
+			requests = append(requests, tmpReq)
+
+			if err != nil {
+				return nil, fmt.Errorf("Can't create the request: %s", err.Error())
+			}
+		}
+	}
+	// If no possible search field, exit
+	if len(requests) == 0 {
+		return nil, fmt.Errorf("%s", errors.New("No compatible hashlookup search field."))
+	}
+
+	for _, req := range requests {
+		// Set basic auth credentials if given
+		if p.source.Access["apiKey"] != "" {
+			req.SetBasicAuth(p.source.Access["apiKey"], "")
+		}
+		req.Header.Add("Content-Type", "application/json")
+		// Declare an HTTP client to execute the request
+		client := http.Client{Timeout: p.source.Timeout}
+		// Send an HTTP using `req` object
+		tmpResp, err := client.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("Can't do an HTTP request: %s", err.Error())
+		}
+
+		tmpBody := &bytes.Buffer{}
+		_, err = tmpBody.ReadFrom(tmpResp.Body)
+		if err != nil {
+			tmpResp.Body.Close()
+			return nil, fmt.Errorf("Can't read an HTTP response: %s", err.Error())
+		}
+		tmpResp.Body.Close()
+		// Check the response
+		if tmpResp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("Bad response StatusCode: %s", tmpResp.Status)
+		}
+		responses = append(responses, tmpBody)
+	}
+
+	return responses, nil
+}
+
+// getChildren connect to the HTTP endpoints and retrieve children
+func (p *plugin) getRelated(relation string, parent string, limit int) (*gabs.Container, error) {
+	fmt.Println(relation)
+
+	// Do everything as float64 as shortcut because it's gabs's default
+	var nbElements, total, spoon float64
+	gabsBuffer := gabs.New()
+	gabsBuffer.Array(relation)
+	var err error
+
+	// How many records we want at once
+	spoon = 99
+	url := ""
+	cursor := "0"
+	total = spoon
+
+	for (nbElements < float64(limit)) && (nbElements < total) {
+		fmt.Println(nbElements)
+		url = p.url + "/" + relation + "/" + parent + "/" + fmt.Sprint(spoon) + "/" + cursor
+		fmt.Println(url)
+		body, err := p.query(url)
+		if err != nil {
+			return nil, fmt.Errorf("Can't query %s", err.Error())
+		}
+		tmp, err := gabs.ParseJSONBuffer(body)
+		if err != nil {
+			return nil, fmt.Errorf("Can't parse: %s", err.Error())
+		}
+		gabsBuffer.Merge(tmp)
+		cursor = tmp.Path("cursor").Data().(string)
+		total, _ = tmp.Path("total").Data().(float64)
+		nbElements += spoon
+	}
+
+	return gabsBuffer, err
+}
+
+func (p *plugin) Stop() error {
+	/*
+	 * STEP 8.
+	 *
+	 * Stop the plugin when main service stops,
+	 * drop all connections correctly
+	 */
+
+	// ctx, cancel := context.WithTimeout(context.Background(), p.source.Timeout)
+	// defer cancel()
+
+	// return p.client.Disconnect(ctx)
```

Comment:
Code left from a template. Can be replaced with just `No error to check, so return nil`